### PR TITLE
III-5529 Add `wfstatus` to granular events

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added
- Added `getModeration()` to calculate `WorkflowStatus` from `CdbXml` in trait `EventFromUDB2`
- Added tests to check getModeration() in `EventFromUDB2`
- test `CdbXml`-files with various `wfstatus`

---
Ticket: https://jira.uitdatabank.be/browse/III-5529

### CODE REVIEW

File: src/Event/Events/EventFromUDB2.php [hunk 1]
```diff
@@ -201,4 +215,28 @@ public function getExternalId(): ?string
...
+
+        if ($wfstatus === 'rejected') {
+            return new Rejected($this->eventId, new StringLiteral('Reason unknown (imported from UiTdatabank v2)'));
+        }
+
+        if ($wfstatus === 'deleted') {
+            return new EventDeleted($this->eventId);
+        }
+
+        return null;
```
Comment (LucWollants) [inline src/Event/Events/EventFromUDB2.php:240]: Which `wfstatus` in XML can not be converted to a moderation event?

File: src/Event/Events/EventFromUDB2.php [hunk 2]
```diff
@@ -201,4 +215,28 @@ public function getExternalId(): ?string
         $eventAsArray = $this->getEventAsArray();
         return $eventAsArray['location'][0]['label'][0]['@attributes']['externalid'] ?? null;
     }
+
+    private function getModeration(?string $wfstatus, ?string $lastUpdated): ?AbstractEvent
```
Comment (LucWollants) [inline src/Event/Events/EventFromUDB2.php:219]: Can these parameters be nullable if an `isset` is used before calling it?

File: src/Event/Events/EventFromUDB2.php (hunk 1 above)
Comment (JonasVHG) [reply src/Event/Events/EventFromUDB2.php:240]: The `wfstatus` could still be `DRAFT` in `CdbXml`, which is not a moderation event in UDB3.

File: src/Event/Events/EventFromUDB2.php (hunk 2 above)
Comment (JonasVHG) [reply src/Event/Events/EventFromUDB2.php]: Indeed, fixed it.

File: src/Event/Events/EventFromUDB2.php (hunk 1 above)
Comment (LucWollants) [reply src/Event/Events/EventFromUDB2.php:240]: Thanks for clarifying, maybe add a small comment for this in the code?

File: src/Event/Events/EventFromUDB2.php [hunk 3]
```diff
@@ -201,4 +215,28 @@ public function getExternalId(): ?string
         $eventAsArray = $this->getEventAsArray();
         return $eventAsArray['location'][0]['label'][0]['@attributes']['externalid'] ?? null;
     }
+
+    private function getModeration(string $wfstatus, string $lastUpdated): ?AbstractEvent
+    {
+        if ($wfstatus === 'readyforvalidation') {
+            return new Published(
+                $this->eventId,
+                \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:s', $lastUpdated, new DateTimeZone('Europe/Brussels')),
```
Comment (LucWollants) [inline src/Event/Events/EventFromUDB2.php:224]: Do we need this explicit hardcoded timezone?

File: src/Event/Events/EventFromUDB2.php (hunk 1 above)
Comment (JonasVHG) [reply src/Event/Events/EventFromUDB2.php:240]: Done.

**Review by LucWollants — APPROVED** [review]

**Pull request merged**